mlflow/sagemaker/__init__.py

```diff
@@ -1355,16 +1355,38 @@ def _get_deployment_config(flavor_name, env_override=None):
     return deployment_config
 
 
+def _truncate_name(name, max_length):
+    # NB: Sagemaker prevents the registration of models and configurations whose names
+    # exceed 63 characters in length. For reference:
+    # https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_Model.html
+    # https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_TransformJob.html
+    # https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_ModelConfiguration.html
+    # This function middle-truncates the name provided to
+    # ensure that the least critical name information is not lost
+    if len(name) <= max_length:
+        return name
+    available_length = max_length - 3
+    start_len = available_length // 2
+    end_len = available_length - start_len
+    return f"{name[:start_len]}...{name[-end_len:]}"
```

Comment:
Can this lead to an undesired model name? What if my model name looks like "hey-please-do-not-truncate-THIS-foo-bar"?

Reply:
It definitely could. However, the current process of aborting deployment because we append a unique uuid to the end of the model name and model config name is even worse. 
With the habits of naming conventions in versioned artifacts, (start of name defines the project, end defines the version of iteration) I felt as though a middle truncation is the least destructive of critical information for a user to see within the sagemaker UI which model and which version of their model is deployed as a service. 
The alternative is to reduce the generated uuid length (before setting the fixed length to 20, on 100 iterations, I found a behavior of generated lengths between 20-22). We could reduce that fixed length number down, to say, 12? And still dramatically reduce the chances of duplicate deployment names. 
What do you think?

Reply:
Got it, can we emit a warning message?

Reply:
Is `.` an allowed character?